src/core/logic.ml

```diff
@@ -405,6 +820,31 @@ module Index = struct
     with
     | _ -> ()
 
+  (* robSecondStore () = ()
+     Store all comptype theorem constants in the `compTTypes' table.
+  *)
+  let robSecondStore () =
+    try
+      List.iter storeCompConst (Cid.Comp.current_entries ())
+    with
+    | _ -> ()
+
+  (* robThirdStore () = ()
+     Store all  comptyp Inductive constants in the `compITypes' table.
+  *)
+  let robThirdStore () =
+    try
+      List.iter storeCompTypConst (Cid.CompTyp.current_entries ())
+    with
+    | _ -> ()
+
+  (* Store all signature constants in their respective tables *)
+  let robAll () =
+    robStore ();
+    robSecondStore ();
+    robThirdStore ()
```

Comment:
Robbing the global stores does not restrict the search space of queries to the declarations in the signature up to the point where the query is declared. This may be addressed when we revisit the implementation of the store.